shuffle/move/examples/main/helpers.ts

```diff
@@ -61,37 +63,104 @@ export function buildScriptFunctionTransaction(
   );
 }
 
-// Example Usage:
-// await DiemHelpers.buildAndSubmitScriptFunctionTransaction("0xE73FFAAB476ED3F57E1A6877F3EE3891", "Foo", "Bar", [], [], 0)
-export async function buildAndSubmitScriptFunctionTransaction(
-  moduleAddress: string,
-  moduleName: string,
-  functionName: string,
-  tyArgs: Seq<DiemTypes.TypeTag>,
-  args: Seq<bytes>,
+// Invokes a script function using the Dev API's signing_message/ JSON endpoint.
+export async function invokeScriptFunction(
+  addressStr: string,
   sequenceNumber: number,
+  privateKeyBytes: Uint8Array,
+  scriptFunction: string,
+  types: string[],
+  // deno-lint-ignore no-explicit-any
+  args: any[],
 ) {
-  const payload: DiemTypes.TransactionPayload = buildScriptFunctionTransaction(
-    moduleAddress,
-    moduleName,
-    functionName,
-    tyArgs,
-    args,
+  privateKeyBytes = normalizePrivateKey(privateKeyBytes);
+  const [sfAddress, sfModule, sfFunction] = splitFullyQualifiedName(
+    scriptFunction,
   );
+  const typeStructs = splitTypes(types);
 
-  // TODO(dimroc) : Help clean this up
-  const shuffleDir = Deno.env.get("SHUFFLE_HOME") || "unknown";
-  const privateKeyPath = path.join(shuffleDir, "accounts/latest/dev.key");
-  const senderAddressPath = path.join(shuffleDir, "accounts/latest/address");
-  const senderAddress = await Deno.readTextFile(senderAddressPath);
-  const fullSenderAddress = "0x" + senderAddress;
-  const privateKeyBytes = await Deno.readFile(privateKeyPath);
-  return await buildAndSubmitTransaction(
-    fullSenderAddress,
-    sequenceNumber,
-    privateKeyBytes,
-    payload,
-  );
+  // deno-lint-ignore no-explicit-any
+  const request: any = {
+    "sender": addressStr,
+    "sequence_number": `${sequenceNumber}`,
+    "max_gas_amount": "1000000",
+    "gas_unit_price": "0",
+    "gas_currency_code": "XUS",
+    "expiration_timestamp_secs": "99999999999",
+    "payload": {
+      "type": "script_function_payload",
+      "module": {
+        "address": sfAddress,
+        "name": sfModule,
```

Comment:
Should we use string type module full name? I mean server side current accepts this struct, but it seems we should keep it simple like struct tag, just accepts "0x1::DiemAccount" string instead of a struct. If you also think that's better, I will make a change in short term. We keep this PR as it is, I will update it when I update server side data structure

Reply:
I'm for the change, but would we need to support both for backwards compatibility? Or can we make the switch now before anyone uses it? I can hold off on this PR until that change is done.

Reply:
sounds good, I'll make change today.